File: 001.js
'use strict';

var sum = function(array) {
    return array.reduce(function(a, b) {
        return a+b;
    });
};

var multiples = function(max) {
  var multiplesOfThreeOrFive = [];
  for (var i = 1; i < max; i++) {
      if ((i % 3 === 0) || (i % 5 === 0)) {
          multiplesOfThreeOrFive.push(i);
      }
  }
  return multiplesOfThreeOrFive;
};

sum(multiples(1000));

File: 001.rb
def sum_multiples(max)
    array = []
    (1..(max-1)).each do |n|
        if (n % 3 == 0) || (n % 5 == 0)
            array.push(n)
        end
    end
    sum = array.reduce(:+)
    return sum
end

sum_multiples(1000)
